src/Surface.cpp

```diff
@@ -65,6 +65,14 @@ Surface::Surface(const int id, const char* name){
 Surface::~Surface() {
   if (_name != NULL)
     delete [] _name;
+
+  if (!_neighbors.empty()){
+    std::map<int, std::vector<Cell*>* >::iterator iter;
```

Comment:
This would be simpler without the ``for`` loop if we simply use the fact that there are only two integer keys for each halfspace:

```C++
if (!_neighbors.empty()) {
  _neighbors[-1].clear();
  _neighbors[+1].clear();
  _neighbors.clear();
}
```